dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/sanitizer/DdUrlSanitizer.kt

```diff
@@ -0,0 +1,41 @@
+package com.datadog.gradle.plugin.internal.sanitizer
+
+import java.util.Locale
+
+/**
+ * It will strip out the user information from the provided URL by applying the following rules:
+ * 1. In case the URL is a valid schema URL : ((ssh|http[s]?):\\) it will strip out the user
+ * information if any from it
+ * 2. In case the URL is a valid schema URL but by mistake the user forgot to add the schema:
+ * [username]:[password]@github.com/(.+).git it will strip out the user information if any and will
+ * return the new url
+ * 3. In case the URL is of a GIT SSH short format: [user@]github.com:(.+).git it will do
+ * nothing assuming that there is no password normally provided in a GIT SSH short format URL.
+ */
+internal class DdUrlSanitizer(
```

Comment:
This name is not really relevant: the `Dd` prefix doesn't provide information, and is useless in the sense that all the code base is in the `dd-sdk-android-gradle-plugin` already. 

If we assume that the `UrlSanitizer` interface aims to "cleanup a url to be usable in DD", then this implementation specific focus is to "strip out the user information from the provided URL" as mentioned above. I'd name this `UserInfoUrlSanitizer` or something similar.